### PR TITLE
Crypto iteration 1

### PR DESCRIPTION
**Pull request by arthurchan35**
hey Brian, /cc @arthurchan35 

we believe you simplified too much so we felt we needed to at least make the verify method secure against timing attacks. 
We will need also different algorithms to be configured, so we can't limit this to https://github.com/liferay-appsec/liferay-portal/pull/245/files#diff-cb365d399b92b47ec702ee0764ba56487be103201e0d1c34fb8e6490b578c07aR36 but we are adding this ability in the next PR.

### CODE REVIEW

File: modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/CryptoHashGeneratorImpl.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/main/java/com/liferay/portal/crypto/hash/internal/CryptoHashGeneratorImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
@csierra

**Comment by csierra** [comment]
ci:forward

**Comment by csierra** [comment]
ci:reevaluate:1192004_7632

**Comment by csierra** [comment]
ci:forward

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
CI was reported back online on 17:30pm PST
Retry ci test at: https://github.com/liferay-appsec/liferay-portal/pull/248

**Pull request closed without merging**